### PR TITLE
chore(inputs): Remove deprecated options from sample config

### PR DESCRIPTION
**Pull request by Hipska**
## Summary

Remove deprecated options from the sample config as they might be confusing and also cleanup some code to not use these options.

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

File: plugins/inputs/consul/consul.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/hashicorp/consul/api"
 
 	"github.com/influxdata/telegraf"
+	telegrafConfig "github.com/influxdata/telegraf/config"
```
Comment (srebhan) [inline plugins/inputs/consul/consul.go:12]: Why do you need an alias? If you really need it the new convention is snake-case i.e.
```suggestion
	telegraf_config "github.com/influxdata/telegraf/config"
```

File: plugins/inputs/consul/consul.go [hunk 2]
```diff
@@ -39,9 +40,21 @@ func (*Consul) SampleConfig() string {
...
+				Notice:    `please update to 'metric_version = 2'`,
+			},
+		)
 	}
 
+	newClient, err := c.createAPIClient()
+	if err != nil {
+		return err
+	}
+	c.client = newClient
```
Comment (srebhan) [inline plugins/inputs/consul/consul.go:56]: Are you 100% sure this does not attempt to create a connection? Otherwise this change is a change in behavior of the plugin. Anyway, the complete change to this file seems unrelated to the PR title. I would vote for a separate PR if you whish to pursue those changes.

File: plugins/inputs/kube_inventory/README.md [hunk 3]
```diff
@@ -61,19 +61,9 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
...
-  ##
-  ## If both of these are empty, we'll use the default serviceaccount:
-  ## at: /var/run/secrets/kubernetes.io/serviceaccount/token
-  ##
-  ## To auto-refresh the token, please use a file with the bearer_token option.
-  ## If given a string, Telegraf cannot refresh the token periodically.
   # bearer_token = "/var/run/secrets/kubernetes.io/serviceaccount/token"
-  ## OR
-  ## deprecated in 1.24.0; use bearer_token with a file
-  # bearer_token_string = "abc_123"
```
Comment (srebhan) [inline plugins/inputs/kube_inventory/README.md:76]: I'm not sure if this deprecation makes a lot of sense actually now that we can use secret-stores... The removal will take away that ability. What do you think?

File: plugins/inputs/modbus/configuration_register.go [hunk 4]
```diff
@@ -279,8 +279,8 @@ func (c *ConfigurationOriginal) validateFieldDefinitions(fieldDefs []fieldDefini
 func (c *ConfigurationOriginal) normalizeInputDatatype(dataType string, words int) (string, error) {
 	if dataType == "FLOAT32" {
 		config.PrintOptionValueDeprecationNotice("input.modbus", "data_type", "FLOAT32", telegraf.DeprecationInfo{
-			Since:     "v1.16.0",
-			RemovalIn: "v1.35.0",
+			Since:     "1.16.0",
+			RemovalIn: "1.36.0",
```
Comment (srebhan) [inline plugins/inputs/modbus/configuration_register.go:283]: Is there any reason to postpone the removal?

File: plugins/inputs/sqlserver/README.md [hunk 5]
```diff
@@ -256,31 +256,6 @@ to use them.
...
-  ## Version 2 is being deprecated, please consider using database_type.
-  ## - PerformanceCounters
-  ## - WaitStatsCategorized
-  ## - DatabaseIO
-  ## - ServerProperties
-  ## - MemoryClerk
-  ## - Schedulers
-  ## - SqlRequests
-  ## - VolumeSpace
-  ## - Cpu
```
Comment (srebhan) [inline plugins/inputs/sqlserver/README.md:270]: Why do you remove this information?

File: plugins/inputs/sqlserver/README.md [hunk 6]
```diff
@@ -341,7 +316,7 @@ the new format. The version 2 queries only report raw metrics, no math has
 been done to calculate deltas. To graph this data you must calculate deltas
 in your dashboarding software.
 
-### Version 1 (query_version=1): This is Deprecated in 1.6, all future development will be under configuration option database_type
+### Version 1 (query_version=1): This is Deprecated in 1.16, all future development will be under configuration option database_type
```
Comment (srebhan) [inline plugins/inputs/sqlserver/README.md:319]: Why do you change the deprecated version?

**Review by srebhan — COMMENTED** [review]
Thank you very much @Hipska for taking the effort cleaning up the configs! I do have some comments/questions in the code...

File: plugins/inputs/consul/consul.go (hunk 2 above)
Comment (Hipska) [reply plugins/inputs/consul/consul.go:56]: Yup, see https://github.com/hashicorp/consul/blob/api/v1.29.4/api/api.go#L674

File: plugins/inputs/consul/consul.go (hunk 1 above)
Comment (Hipska) [reply plugins/inputs/consul/consul.go:12]: There is a conflict in `createAPIClient` method otherwise.

File: plugins/inputs/kube_inventory/README.md (hunk 3 above)
Comment (Hipska) [reply plugins/inputs/kube_inventory/README.md:76]: `bearer_token_string` is a regular string, not a secret. Also, it doesn't refresh if it would..

File: plugins/inputs/modbus/configuration_register.go (hunk 4 above)
Comment (Hipska) [reply plugins/inputs/modbus/configuration_register.go]: Good catch, must be a typo. Fixed in b198412

File: plugins/inputs/sqlserver/README.md (hunk 6 above)
Comment (Hipska) [reply plugins/inputs/sqlserver/README.md:319]: I just fixed the docs to what the code is saying.

File: plugins/inputs/sqlserver/README.md (hunk 5 above)
Comment (Hipska) [reply plugins/inputs/sqlserver/README.md:270]: Both options are removed and are still described what they provide in the README, this doesn't need to exist in the config.

File: plugins/inputs/kube_inventory/README.md (hunk 3 above)
Comment (srebhan) [reply plugins/inputs/kube_inventory/README.md:76]: But we can convert the string to a secret easily. If we now remove it, we need to add it back in later...

File: plugins/inputs/consul/consul.go (hunk 2 above)
Comment (srebhan) [reply plugins/inputs/consul/consul.go:56]: Anyway, please move this to an own PR so we can revert it independently if we run into trouble.

File: plugins/inputs/consul/consul.go (hunk 1 above)
Comment (srebhan) [reply plugins/inputs/consul/consul.go:12]: Then please use snake case.

File: plugins/inputs/kube_inventory/README.md (hunk 3 above)
Comment (Hipska) [reply plugins/inputs/kube_inventory/README.md:76]: I'm not removing any functionality, just removing deprecated settings from the sample configs.. 🤷

File: plugins/inputs/consul/consul.go (hunk 2 above)
Comment (Hipska) [reply plugins/inputs/consul/consul.go]: Okay, reverted in c29243cd4dd05f96daf3986b5f9f5c783eac1c31.

**Review by srebhan — APPROVED** [review]
Thanks @Hipska!

File: plugins/inputs/kube_inventory/README.md (hunk 3 above)
Comment (srebhan) [reply plugins/inputs/kube_inventory/README.md:76]: I don't want to discuss this and will simply add the same setting back later...

**Review by DStrand1 — APPROVED** [review]
